packages/core/src/dialog/DialogContent.tsx

```diff
@@ -37,14 +42,35 @@ export const DialogContent = forwardRef<HTMLDivElement, DialogContentProps>(
       window: targetWindow,
     });
 
+    useEffect(() => {
+      const checkOverflow = () => {
+        if (!divRef.current) return;
+        setIsOverflowing(
+          divRef.current.scrollHeight > divRef.current.clientHeight,
+        );
+      };
+      checkOverflow();
+
+      window.addEventListener("resize", checkOverflow);
+      return () => {
+        window.removeEventListener("resize", checkOverflow);
+      };
+    }, [children]);
+
     return (
       <>
         <div className={clsx({ [withBaseName("scroll")]: scrollTop > 0 })} />
         <div
-          className={clsx(withBaseName(), className)}
+          className={clsx(
+            withBaseName(),
+            {
+              [withBaseName("overflow")]: isOverflowing,
```

Comment:
Does `isOverflowing` serves the same condition as `scrollTop > 0` above ?

Reply:
not quite, isOverflowing checks if the content wraps into overflow since loading, scrollTop checks where the scroll is to add the top shadow... but I'll see if i can do without it

Reply:
couldnt get rid of, but I optimised the  scrolled check